test/test-exception.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+// Testcase to produce NodeReport on uncaught exception
+if (process.argv[2] === 'child') {
+  require('../').setEvents("exception");
+
+  function myException(request, response) {
+    throw new UserException('*** exception.js: testcase exception thrown from myException()');
+  }
+
+  function UserException(message) {
+    this.message = message;
+    this.name = "UserException";
+  }
+
+  myException(); 
+} else {
+  const tap = require('tap');
+  const common = require('./common.js');
+  const fs = require('fs');
+  const spawn = require('child_process').spawn;
+
+  var child = spawn(process.execPath, [__filename, 'child']);
+  child.on('exit', function(code, signal) {
+    tap.plan(3);
+    tap.notEqual(code, 0, 'Process should not exit cleanly');
```

Comment:
Well that didn't work (tried Node v4.6.1, v6.9.1 on x64 Linux):
```
-bash-4.2$ ./node_modules/.bin/tap --reporter=tap test/test-exception.js
TAP version 13
# Subtest: test/test-exception.js
    1..3
    not ok 1 - Process should exit with status 1
      ---
      found: null
      wanted: 1
      compare: ===
      at:
        line: 26
        column: 9
        file: test/test-exception.js
        type: ChildProcess
      stack: |
        ChildProcess.<anonymous> (test/test-exception.js:26:9)
      source: |
        tap.equal(code, 1, 'Process should exit with status 1');
      ...
    ok 2 - found report NodeReport.20161102.160920.8809.001.txt
    # Subtest: validating NodeReport.20161102.160920.8809.001.txt
        1..4
        ok 1 - Report contains NodeReport section
        ok 2 - Report contains JavaScript Stack Trace section
        ok 3 - Report contains JavaScript Heap section
        ok 4 - Report contains System Information section
    ok 3 - validating NodeReport.20161102.160920.8809.001.txt # time=15.683ms
    # failed 1 of 3 tests
    # time=763.704ms
not ok 1 - test/test-exception.js # time=971.072ms
  ---
  timeout: 30000
  results:
    ok: false
    plan:
      start: 1
      end: 3
    pass: 2
    fail: 1
    count: 3
  arguments:
    - test/test-exception.js
  file: test/test-exception.js
  exitCode: 1
  command: /dev/shm/usenode.riclau/node-v4.6.1-linux-x64/bin/node
  ...

1..1
# failed 1 of 1 tests
# time=1005.135ms
-bash-4.2$
```
and
```
-bash-4.2$ node test/test-exception.js child ; echo $?

Writing Node.js report to file: NodeReport.20161102.161145.9112.001.txt
Node.js report completed
Uncaught #<UserException>

FROM
myException (/home/users/riclau/sandbox/github/nodereport/test/test-exception.js:8:5)
Object.<anonymous> (/home/users/riclau/sandbox/github/nodereport/test/test-exception.js:16:3)
Module._compile (module.js:409:26)
Object.Module._extensions..js (module.js:416:10)
Module.load (module.js:343:32)
Function.Module._load (module.js:300:12)
Function.Module.runMain (module.js:441:10)
startup (node.js:139:18)
node.js:974:3
Illegal instruction (core dumped)
132
-bash-4.2$
```
Guess this is a bug? cc @rnchamberlain